### PR TITLE
[full-ci] TypeScript almost everything

### PR DESCRIPTION
**Pull request by dschmidt**
## Description
While #8235 focused on Vue components, this converts basically all other files currently still used to TypeScript - with a few exceptions: acceptance tests, docs, a few legacy (config) files for design system or web-app-skeleton

## Related Issue
- Follow Up to #8235 

## Motivation and Context
Get type safety to avoid bugs (and find existing ones), reduce mental load by making sure all files are written in the same language, make cut pasting easier, ....

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: packages/web-app-files/tests/unit/helpers/breadcrumbs.spec.ts [hunk 1]
```diff
@@ -14,15 +18,18 @@ describe('builds an array of breadcrumbitems', () => {
...
       JSON.stringify({
-        allowContextActions: undefined,
-        onClick: () => jest.fn(() => {}),
-        text: undefined
+        allowContextActions: true,
+        text: 'bar',
+        onClick: () => undefined
       })
     )
   })
```
Comment (dschmidt) [inline packages/web-app-files/tests/unit/helpers/breadcrumbs.spec.ts:35]: IMHO this test was completely broken as `breadCrumbsFromPath` was not spread into `concatBreadcrumbs` which just accepts a single rest args parameter.

File: packages/web-runtime/src/services/clientRegistration.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ async function post(url, data) {
 }
 
 export async function registerClient(openIdConfig) {
-  const clientData = sessionStorage.getItem('dynamicClientData')
+  const clientData = JSON.parse(sessionStorage.getItem('dynamicClientData'))
```
Comment (dschmidt) [inline packages/web-runtime/src/services/clientRegistration.ts:32]: I believe this was an actual bug, I have no idea how to test the whole feature, which is why I did not add a Changelog entry.

**Comment by ownclouders** [comment]
Results for <strong>acceptance oC10</strong> https://drone.owncloud.com/owncloud/web/31928/27/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingInternalGroupsToRoot-shareWithGroups_feature-L21.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>acceptance oCIS</strong> https://drone.owncloud.com/owncloud/web/31929/62/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingInternalGroups-shareWithGroups_feature-L19.png
  


</p></details>

File: packages/web-runtime/tests/unit/helpers/theme.spec.ts [hunk 3]
```diff
@@ -21,28 +17,28 @@ describe('theme loading and error reporting', () => {
   })
 
   it('should load the default theme if location is not found', async () => {
-    fetch.mockResponse(new Error(), { status: 404 })
+    ;(fetch as FetchMock).mockResponse(new Error() as any, { status: 404 })
```
Comment (dschmidt) [inline packages/web-runtime/tests/unit/helpers/theme.spec.ts:20]: I found no method with a signature that allows me to pass an error and a status ... hence I need to cast error :-\

**Review by JammingBen — APPROVED** [review]

**Pull request merged**